### PR TITLE
[dash-p4] Refactor routing type and routing actions in P4 to make it align more explicitly with the DASH pipeline design.

### PR DESCRIPTION
**Pull request by r12f**
This commit refactors the table actions in outbound pipeline to make it align more explicitly to how DASH pipeline is modeled today. 

The change includes:
- Extract the table actions related to routing types out of the control block into a separate file.
- Split the routing type actions into a list of routing actions.

For how routing types and routing actions work, please refer to the [SONiC DASH HLD](https://github.com/sonic-net/DASH/blob/main/documentation/general/dash-sonic-hld.md).

This commit doesn't change any SAI header definitions and libsai (except a few action id change, due to moving actions):
image

image

### CODE REVIEW

File: dash-pipeline/bmv2/stages/routing_action_apply.p4 [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#define _DASH_STAGE_ACTION_APPLY_P4_
+
+#include "../routing_actions/routing_actions.p4"
+
+control routing_action_apply(
+    inout headers_t hdr,
+    inout metadata_t meta)
+{
+    apply {
+        do_action_static_encap.apply(hdr, meta);
```
[the remark this answers is not in the document]
Comment (r12f) [reply dash-pipeline/bmv2/stages/routing_action_apply.p4:11]: They are actually independent, as they never touch the same field, so order doesn't matter : D

[the remark this answers is not in the document]
Comment (r12f) [reply dash-pipeline/bmv2/stages/routing_action_apply.p4:11]: with this design, theoretically, all routing actions can be performed in parallel, if the underlying ASIC supports.

[the remark this answers is not in the document]
Comment (r12f) [reply dash-pipeline/bmv2/stages/routing_action_apply.p4:11]: Actually, you are right. The length and checksum are the ones needs to be fixed for parallel execution. Will fix the order.

[the remark this answers is not in the document]
Comment (r12f) [reply dash-pipeline/bmv2/stages/routing_action_apply.p4]: This is fixed now.

**Review by marian-pritsak — APPROVED** [review]

**Comment by r12f** [comment]
woot! thanks Marian!

**Pull request merged**